src/cs1/linked-lists/linked-list-basics.md

```diff
@@ -0,0 +1,401 @@
++------+    +------+    +------+    +------+    +------+
+    ^
+    |
++----------+
+| tempHead |
++----------+
+```
+
+We can utilize the ability to use `tempHead->next` to check the value in front of us. Once we find the first value that is greater than `13` we know that we make `13` the next pointer of `tempHead`.
+
+
+```
++------+    +------+    +------+    +------+
+|  11  | -> |  15  | -> |  18  | -> | NULL |
++------+    +------+    +------+    +------+
+    ^
+    |
++----------+
+| tempHead |
++----------+
+```
+
+Since we know that `tempHead->next` = `15`, we will add `13` into the list 
+
+```
++------+    +------+    +------+    +------+
+|  11  | -| |  15  | -> |  18  | -> | NULL |
++------+  | +------+    +------+    +------+
+    ^     |      ^
+    |     |      |-----     
+    |     ------      |
++----------+   |   +------+
+| tempHead |   |-> |  13  |
++----------+       +------+
+```
+
+We set `13` next value as `15` and then we set `tempHead`'s next value to 13. Now we've inserted `13` in the correct place within the list
+
+It's clear to see that at WORST we might have to traverse the whole list, which makes this an `O(n)` operation.
+
+```c
+node* addInOrder(node* head, int val){
+    node* toAdd = makeNode(val);
+    //We just created a new list
+    if(head==NULL)
+        return toAdd;
+    
+    //If the value we want to add is less than the head our new node is the head
+    if(val<head->data){
+        toAdd->next = head;
+        return toAdd;
+    }
+    //We know now that head must be < val, so we need to continually check temp->next elements
+    node* temp = head;
+    
+    //Find the spot to place our new node
+    while(temp->next!=NULL && temp->next->data<val)
+        temp = temp->next;
+        
+    //After this, temp must be greater than our val
+    //Inserting toAdd between temp and temp->next
+    toAdd->next = temp->next;
+    temp->next = toAdd;
+    //Return the new head of the list
+    return head;
+}
+```
+
+# Delete
+
+Deleting an element from a Linked List will be a lot like adding in order. We will need to keep track of the node that comes right before the node we want to delete. 
+
+Once we "patch over" the deleted node, then we can free it.
+
+Let's delete `15` from the list
+
+```
++------+
+| head |
++------+
++------+    +------+    +------+    +------+    +------+
+|  10  | -> |  11  | -> |  15  | -> |  18  | -> | NULL |
++------+    +------+    +------+    +------+    +------+
+```
+
+Just like before, we will need a temporary node to traverse the rest of the list. Once we find the node RIGHT BEFORE `15`, then we can perform the "patch over" and deletion of `15`.
+
+```
++------+    +------+    +------+    +------+
+|  11  | -> |  15  | -> |  18  | -> | NULL |
++------+    +------+    +------+    +------+
+   ^
+   |
++----------+
+| tempHead |
++----------+
+```
+
+Once we are here we will save `15` in another temporary node, then "patch over", then free `15.
```

Comment:
Same as earlier, should this be a temporary node or a temporary pointer? Since we are working with these new concepts in linked lists, that distinction can be very important! (From experience as a TA you can never be too careful with thinking that a student will assume to know what you mean.)

Reply:
Perhaps for `NULL` we can just use an `X` like I used for the binary trees?